File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        ans=[0]*len(nums)
        left=0
        right=len(nums)-1
        index=len(nums)-1
        while left<=right:
            if nums[left]**2>nums[right]**2:
                ans[index]=nums[left]**2
                left+=1
            else:
                ans[index]=nums[right]**2
                right-=1
            index-=1
        return ans



        '''
        for index in range(len(nums)):
            nums[index]=nums[index]**2
        nums.sort()
        return nums
        '''

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        
        int[] ans=new int[nums.length];

        int k=nums.length-1;
        int left=0,right=nums.length-1;

        while(left<=right){
            int ls=nums[left]*nums[left];
            int rs=nums[right]*nums[right];

            if(ls>rs){
                ans[k]=ls;
                k--;
                left++;
            }
            else{
                ans[k]=rs;
                k--;
                right--;
            }

        }

        return ans;

    }
}